src/OrleansRuntime/Streams/QueueBalancer/StreamQueueBalancerFactory.cs

```diff
@@ -21,7 +22,7 @@ internal class StreamQueueBalancerFactory
         /// <param name="siloMaturityPeriod">Maturity Period of a silo for queue rebalancing purposes</param>
         /// <returns>Constructed stream queue balancer</returns>
         public static IStreamQueueBalancer Create(
-            StreamQueueBalancerType balancerType,
+            string balancerType,
             string strProviderName,
             ISiloStatusOracle siloStatusOracle,
```

Comment:
see my latest commit. We decided to shift from "configure balancer through a factory" to "configure balancer through its init method", as a more straight-forward alternative. so this factory interface is removed. 
Concern for a keyed factory is, it  split configuration into two places: factory class and keyed service config in DI, which IMO is not an very comprehensive pattern to users. And the constraits we put on the interface is not enough to avoid bad user code.  It's easy for user to misunderstand this feature as "inject an factory as a singleton and do a case-switch inside its create method to generate the correct balancer"